pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -372,23 +374,56 @@ func (gb *GraphBuilder) addDependentToOwners(n *node, owners []metav1.OwnerRefer
 			gb.attemptToDelete.Add(ownerNode)
 		} else if !addedToAttemptToDelete && ownerNode.isObserved() {
 			// The owner node has been observed via an informer
```

Comment:
the diff is unpleasant because this changed the order to check for namespace mismatch first (since we record an event for that scenario and can surface more useful info to the user)